File: algorithm/python/longest_substring_without_repeating.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = 0
        max_count = 0
        tmp = []
        for i in range(len(s)):
            if s[i] in tmp:
                count = count - tmp.index(s[i]) 
                tmp = tmp[(tmp.index(s[i])+1):]
                tmp.append(s[i])
            else:
                count += 1
                if max_count < count:
                    max_count = count
                tmp.append(s[i])
        return max_count
                

File: algorithm/golang/longest_substring_without_repeating.go
func lengthOfLongestSubstring(s string) int {
    var count = 0
    var max_count = 0
    var tmp []string
    for i := 0; i < len(s) ; i++ {
        is_in := false
        for j := 0 ; j < len(tmp) ; j++ {
            if tmp[j] == string(s[i]){
                count = count - j
                tmp = tmp[j+1:]
                tmp = append(tmp, string(s[i]))
                is_in = true
                break
            }
        }
        if is_in == false{
            count ++
            if max_count < count{
                max_count = count
            }
            tmp = append(tmp, string(s[i]))
        }
        
    }
    return max_count
}